### ISSUE TITLE
Hashing content attributes

### ISSUE THREAD

**Issue by furudean** [posthtml/posthtml-hash#48]
With the current behaviour of only checking attributes `href` and `src` the following snippet will be ignored and just output as it is.

```html
<meta property="og:image" content="https://type-kana.cass.moe/assets/social.[hash].png">
```

It would be nice to be able to pick which attributes to look in. Alternatively just expanding the regex to also check in `content` attributes. I think it's a pretty common use case since these tags are used for link previews.

**Comment by furudean** [comment]
I guess this is more of a question, but would a full URL work like in my example? Or does it have to be relative to root? If not then I guess that would be a separate feature request

**Comment by metonym** [comment]
A remote URL wouldn't work because the URL would also need the hash in its filename. For remote URLs, you could append "?v=1" to the URL and increment or change the search params. I believe GitHub does this.

**Comment by furudean** [comment]
It's technically not a remote URL. The file is on the same origin, and these files are included in the bundle. The OpenGraph protocol just requires a full url.

**Comment by metonym** [comment]
I see. This is a compelling use case though.

**Comment by furudean** [comment]
Having support to hash these kinds of files would be very useful to have in `posthtml-hash`

**Comment by metonym** [comment]
If the file is local, I agree that it's within the scope of this library.

We could explore adding an attribute matcher for "content." We'll need to check if the URL is absolute and only use the URL path.

**Comment by furudean** [comment]
Is there any sense to pass a prefix to use for the `content` attribute matcher?

This could allow you to specify the domain + path to search in without having to add logic that tries to match the protocol, subdomain, domain name etc in posthtml-hash.

**Comment by metonym** [comment]
Nice suggestion. That would make it easier.

```js
posthtmlHash({ prefixUrl: "https://type-kana.cass.moe/" });
```

1) `https://type-kana.cass.moe/assets/social.[hash].png` is matched
2) remove `prefixUrl` in matched URL --> `assets/social.[hash].png`
3) check if `assets/social.[hash].png` exists locally, relative to the users current working directory
4) ...


---

One side note though. The hashed static asset will be renamed, which means you will need a separate build process that copies the original asset ("social.[hash].png") to the assets directory.

**Comment by furudean** [comment]
Nice!

> One side note though. The hashed static asset will be renamed, which means you will need a separate build process that copies the original asset ("social.[hash].png") to the assets directory.

You can use something like [bengsfort/rollup-plugin-copy-assets](https://github.com/bengsfort/rollup-plugin-copy-assets) to include arbitrary files in the bundle. I tried this with paths that are relative to the current origin and it worked fine:

```html
<link rel="icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.556e6061.png">
<link rel="icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.fe478c7a.png">
```

It's worth adding a note about this in the documentation I think, so people don't get confused.

**Comment by metonym** [comment]
I welcome any feedback for PR #49 

From an API design perspective, I went with an optional `transformPath` method because it gives the consumer more control in manipulating/filtering matching attribute values.

**Comment by metonym** [comment]
@c-bandy released in [v1.2.0](https://github.com/posthtml/posthtml-hash/releases/tag/v1.2.0).

LMK if there are any issues.

**Comment by furudean** [comment]
I think I've spotted something odd:

```html
<meta property="og:image" content="https://type-kana.cass.moe/assets/social.[hash].png">
<meta property="twitter:image" content="https://type-kana.cass.moe/assets/social.[hash].png">
```

becomes

```html
<meta property="og:image" content="https://type-kana.cass.moe/assets/social.9d1fc19f.png">
<meta property="twitter:image" content="https://type-kana.cass.moe/assets/social.[hash].png">
```

Despite referencing the exact same path, it seems the second one is ignored here.

**Comment by metonym** [comment]
Hot fixed in [v1.2.2](https://github.com/posthtml/posthtml-hash/releases/tag/v1.2.2)

**Comment by furudean** [comment]
Works just fine for me now! Thanks! 

Closing this issue.
